### ISSUE TITLE
Modal Buttons

### ISSUE THREAD

**Issue by angelina-smith** [GSA/px-benefit-finder#1270]
### What's expected

Location: Modal
Summary: Both "Review selections" and "See results" options should be coded as buttons (to match their correct visual stylings as buttons).

From USA.gov's accessibility audit:
> "Since the user is submitting data and not navigating to a different location the link should be coded as an HTML button. Alternately, add the ARIA role="button" so that screen readers announce it as a button. Additional JavaScript will also need to be added so that keyboard users can activate the button using both the Enter and Spacebar keys."

![image](https://github.com/GSA/px-benefit-finder/assets/142342010/fa2f63de-4159-4170-a87b-38144f114e22)

### What happened

From USA.gov's accessibility audit:
> Review selection and See results are visibly presented as buttons but they are coded as links.

### How to replicate

NVDA x Chrome. Screenshot that follows is from NVDA Speech Viewer which reads both options as links.
![image](https://github.com/GSA/px-benefit-finder/assets/142342010/b728ccc2-e2fe-48f6-97ec-a98244988b27)

**Comment by angelina-smith** [comment]
Confirmed this is acting as expected in Dev.
